Source/PINDiskCache.m

```diff
@@ -451,10 +455,14 @@ - (void)_locked_initializeDiskProperties
             [_sizes setObject:fileSize forKey:key];
             byteCount += [fileSize unsignedIntegerValue];
         }
+        
+        [_knownKeys addObject:key];
```

Comment:
@garrettmoon one suggestion that would improve performance: use a single NSMutableDictionary, with a C struct for the date & byte size.

The struct would contain an NSInteger byteCount (with a sentinel value of -1 if not set), an NSTimeInterval or CFAbsoluteTime for the date. NSValue could wrap it with sizeof(CacheEntryMetadataStruct).

The benefits:
- Replace setting & accessing 2x NSDictionaries and 1x NSSet with just 1 dict.
- Extensible for future metadata storage
- State management a bit more clear and centralized; especially, only one data structure access to lock.

We should definitely land this PR anyway, but figured it was worth a mention!

Reply:
I like that idea!, but I'm not sure about the struct, I think a custom object like PINDiskMetadata would be better and more inline with the objective-c-y nature.